### PR TITLE
Instruction to install mmdet and mmcv-full, as_pandas for detection, inference tutorial with tiny motorbike/voc/coco, save_path issue

### PR DESCRIPTION
**Pull request by yongxinw**
*Issue #, if available:*

*Description of changes:* Instruction to install mmdet and mmcv-full, as_pandas for detection, inference tutorial with tiny motorbike/voc/coco, save_path issue

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/tutorials/multimodal/object_detection/inference/detection_inference_coco.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
+# AutoMM Detection - Inference with Pretrained Faster R-CNN on VOC Format Dataset
+:label:`sec_automm_detection_infer_coco`
+
+In this section, we show an example to run inference COCO dataset in COCO Format.
```
Comment (bryanyzhu) [inline docs/tutorials/multimodal/object_detection/inference/detection_inference_coco.md:4]: I think the title, label, the introduction is a bit conflict. 

1. In the title, it says VOC format data, so I assume it is either VOC dataset or some other customized dataset.
2. Then in the label, it says infer_coco
3. In the introduction, it says run inference on COCO in COCO format, which is different from the title.
4. The list below is also a bit confusing, sometimes VOC, sometimes COCO.
5. Then I read the tutorial, I think it is on COCO dataset, but then in the end when the results are shown, the classes are 20 from VOC. 

Can you do a bit rewording here to reduce ambiguity? Thank you.

File: docs/tutorials/multimodal/object_detection/inference/detection_inference_coco.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+```python
+pred = predictor.predict(test_path)
+```
+
+To run inference and save results, run the following:
+```python
+pred = predictor.predict(test_path, save_results=True)
+```
+Currently, we convert the results to a pandas `DataFrame` and save into a `.txt` file.
```
Comment (bryanyzhu) [inline docs/tutorials/multimodal/object_detection/inference/detection_inference_coco.md:79]: If we are not specifying `as_pandas=True`, I think what we have is just list of lists, right? If we call it `pandas DataFrame`, then later when we specify `as_pandas=True`, what will be the difference?

File: docs/tutorials/multimodal/object_detection/inference/detection_inference_coco.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+df = from_coco_or_voc(test_path)[:10][["image"]]
+
+pred = predictor.predict(df)
+
+visualized = visualize_detection(
+    pred=pred,
+    data=df,
+    detection_classes=predictor.get_predictor_classes(),
+    conf_threshold=conf_threshold,
+    visualization_result_dir=visualization_result_dir,
```
Comment (bryanyzhu) [inline docs/tutorials/multimodal/object_detection/inference/detection_inference_coco.md:142]: `visualization_result_dir` is where the machine-labeled images will go?

File: docs/tutorials/multimodal/object_detection/inference/detection_inference_coco.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+## Running inference
+To run inference, perform:
+
+```python
+pred = predictor.predict(test_path)
+```
+
+To run inference and save results, run the following:
+```python
+pred = predictor.predict(test_path, save_results=True)
```
Comment (bryanyzhu) [inline docs/tutorials/multimodal/object_detection/inference/detection_inference_coco.md:77]: Let's mention by default, we don't save results. But if you want results, you can set `save_results=True`, and you will get a .txt file looking like xxx.

File: docs/tutorials/multimodal/object_detection/inference/detection_inference_quick_start.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+zip_file = "s3://automl-mm-bench/object_detection_dataset/tiny_motorbike_coco.zip"
+```
+
+Now let's download the dataset
+```python
+download_dir = "./tiny_motorbike_coco"  # specify a target download dir to store this dataset
+
+load_zip.unzip(zip_file, unzip_dir=download_dir)
+data_dir = os.path.join(download_dir, "tiny_motorbike")
+train_path = os.path.join(data_dir, "Annotations", "coco_trainval.json")
```
Comment (bryanyzhu) [inline docs/tutorials/multimodal/object_detection/inference/detection_inference_quick_start.md:25]: why coco_trainval and coco_test for tiny_motorbike?

File: docs/tutorials/multimodal/object_detection/inference/detection_inference_quick_start.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+## Reading results
+The returned value `pred` is a `list` and has the following dimensions:
+
+```
+[num_images, num_total_classes, num_detections_per_class, 5]
+```
+
+where 
+- `num_images` is the number of images used to run inference on. 
+- `num_total_classes` is the total number of classes. In this example, `num_total_classes = 20` for VOC dataset
```
Comment (bryanyzhu) [inline docs/tutorials/multimodal/object_detection/inference/detection_inference_quick_start.md:103]: classes=20?

File: multimodal/src/autogluon/multimodal/predictor.py [hunk 7]
```diff
@@ -423,6 +423,12 @@ def __init__(
         self._model_loaded = False  # Whether the model has been loaded
         self._matcher = None
 
+        if self._problem_type == OBJECT_DETECTION:
+            warnings.warn(
+                "Running object detection. Make sure that you have installed mmdet and mmcv-full, "
+                "by running 'pip install mmdet', and 'mim install mmcv-full'"
```
Comment (bryanyzhu) [inline multimodal/src/autogluon/multimodal/predictor.py:429]: Let's put `mim install mmcv-full` first, then `pip install mmdet`

**Review by bryanyzhu — COMMENTED** [review]
Thank you for the tutorials, I left some comments on potential title/content mismatch.

File: docs/tutorials/multimodal/object_detection/inference/detection_inference_coco.md (hunk 3 above)
Comment (yongxinw) [reply docs/tutorials/multimodal/object_detection/inference/detection_inference_coco.md:142]: yea this is the directory to save the machine-labeled images

File: docs/tutorials/multimodal/object_detection/inference/detection_inference_quick_start.md (hunk 5 above)
Comment (yongxinw) [reply docs/tutorials/multimodal/object_detection/inference/detection_inference_quick_start.md:25]: I think it was named coco for its format. @FANGAreNotGnu do you have any insights into this?

File: docs/tutorials/multimodal/object_detection/inference/detection_inference_quick_start.md (hunk 6 above)
Comment (yongxinw) [reply docs/tutorials/multimodal/object_detection/inference/detection_inference_quick_start.md:103]: yea for VOC dataset, there is 20 classes in total.

File: multimodal/src/autogluon/multimodal/predictor.py (hunk 7 above)
Comment (yongxinw) [reply multimodal/src/autogluon/multimodal/predictor.py:429]: ok will do

File: docs/tutorials/multimodal/object_detection/inference/detection_inference_coco.md (hunk 1 above)
Comment (yongxinw) [reply docs/tutorials/multimodal/object_detection/inference/detection_inference_coco.md:4]: Yea I think I've copied things around and might have messed up with the titles!

File: docs/tutorials/multimodal/object_detection/inference/detection_inference_coco.md (hunk 2 above)
Comment (yongxinw) [reply docs/tutorials/multimodal/object_detection/inference/detection_inference_coco.md:79]: Yes if `as_pandas` is not specified, we'll have a list of lists as returned values. 

if `as_pandas=True`, then we'll have the pandas DataFrame as returned values, where there are 2 columns, column1 is imagenames, column2 is list of bounding boxes.

File: docs/tutorials/multimodal/object_detection/inference/detection_inference_quick_start.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
+# AutoMM Detection - Inference with Pretrained Faster R-CNN on VOC Format Dataset
```
Comment (FANGAreNotGnu) [inline docs/tutorials/multimodal/object_detection/inference/detection_inference_quick_start.md:1]: Later we can merge this into our quick start tutorial.

**Review by bryanyzhu — APPROVED** [review]
LGTM, with one comment on changing the name for tiny_motor dataset to `trainval_cocoformat.json`

File: docs/tutorials/multimodal/object_detection/inference/detection_inference_quick_start.md (hunk 8 above)
Comment (yongxinw) [reply docs/tutorials/multimodal/object_detection/inference/detection_inference_quick_start.md]: yea for sure

**Review by FANGAreNotGnu — APPROVED** [review]
The code part LGTM. For the tutorials I think we'll proofread all of them together next week.

**Pull request merged**